src/pages/index.js

```diff
@@ -0,0 +1,329 @@
+  modal.classList.remove("modal_is-opened");
+  document.removeEventListener("keydown", handleCloseModalByEsc);
+  modal.removeEventListener("mousedown", handleCloseModalOverlayClick);
+}
+
+function handleEditFormSubmit(evt) {
+  // const formData = get;
+
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = editModalNameInput.value;
+      profileDescription.textContent = editModalDescriptionInput.value;
+      closeModal(editModal, settings);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitBtn.textContent = "save";
+    });
+}
+
+function handleCloseModalOverlayClick(e) {
+  if (e.target.classList.contains("modal")) closeModal(e.target);
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+
+  const nameInputValue = cardNameInput.value;
+  console.log(nameInputValue);
+
+  const linkInputValue = cardModalLinkInput.value;
+  console.log(linkInputValue);
+
+  const data = {
+    name: cardNameInput.value,
+    link: cardModalLinkInput.value,
+  };
+
+  api
+    .addCard(data)
+    .then((res) => {
+      const cardEl = getCardElement(res);
+      cardsList.prepend(cardEl);
+    })
+    // .catch((error) => {
+    // console.error("error adding card:", error);
+    // });
+    .catch(console.error)
+    .finally(() => {
+      submitBtn.textContent = "save";
+    });
+
+  closeModal(cardModal);
+
+  evt.target.reset();
+  disableButton(cardSubmitButton, settings);
+  cardForm.reset();
+}
+
+profileEditButton.addEventListener("click", () => {
+  editModalNameInput.value = profileName.textContent;
+  editModalDescriptionInput.value = profileDescription.textContent;
+  resetValidation(
+    editFormElement,
+
+    [editModalNameInput, editModalDescriptionInput],
+    settings
+  );
+  openModal(editModal, settings);
+});
+editModalCloseButton.addEventListener("click", () => {
+  closeModal(editModal, settings);
+});
+
+editFormElement.addEventListener("submit", handleEditFormSubmit);
+cardForm.addEventListener("submit", handleAddCardSubmit);
+
+cardModalButton.addEventListener("click", () => {
+  openModal(cardModal, settings);
+});
+
+cardModalCloseButton.addEventListener("click", () => {
+  closeModal(cardModal, settings);
+});
+
+avatarForm.addEventListener("submit", handleAvatarSubmit);
+
+avatarModalButton.addEventListener("click", () => {
+  openModal(avatarModal, settings);
+});
+
+avatarModalCloseButton.addEventListener("click", () => {
+  closeModal(avatarModal, settings);
+});
+
+deleteForm.addEventListener("submit", handleDeleteSubmit);
+
+api.getInitialCards().then((cards) => {
+  cards.forEach((item) => {
+    const cardElement = getCardElement(item);
+    cardsList.append(cardElement);
+  });
+});
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**